weave/tests/test_dataset.py

```diff
@@ -0,0 +1,50 @@
+import pandas as pd
+
+import weave
+
+
+def test_dataset_add():
+    ds = weave.Dataset(rows=[{"a": 1, "b": 2}])
+    ds.add({"a": 3, "b": 4})
+    assert ds.rows == [{"a": 1, "b": 2}, {"a": 3, "b": 4}]
+
+
+def test_dataset_remove():
+    ds = weave.Dataset(rows=[{"a": 1, "b": 2}, {"a": 3, "b": 4}])
+    ds.remove(0)
+    assert ds.rows == [{"a": 3, "b": 4}]
+
+
+def test_dataset_to_frame():
+    ds = weave.Dataset(rows=[{"a": 1, "b": 2}, {"a": 3, "b": 4}])
+    df = ds.to_pandas()
+    assert df.equals(pd.DataFrame({"a": [1, 3], "b": [2, 4]}))
+
+
+def test_gotten_dataset_add(client):
+    ds = weave.Dataset(rows=[{"a": 1, "b": 2}, {"a": 3, "b": 4}])
+    ref = weave.publish(ds)
+
+    ds2 = ref.get()
+    ds2.add({"a": 5, "b": 6})
+    assert ds2.rows == [{"a": 1, "b": 2}, {"a": 3, "b": 4}, {"a": 5, "b": 6}]
+
+
+def test_gotten_dataset_remove(client):
+    # TODO: sqlite server path is broken
+    ds = weave.Dataset(rows=[{"a": 1, "b": 2}, {"a": 3, "b": 4}])
+    ref = weave.publish(ds)
+
+    ds2 = ref.get()
+    ds2.remove(0)
+    assert ds2.rows == [{"a": 3, "b": 4}]
+
+
+def test_gotten_dataset_to_frame(client):
```

Comment:
i think there are some key tests missing:

```
ds = weave.Dataset(rows=[{"a": 1, "b": 2}, {"a": 3, "b": 4}])
ref = weave.publish(ds)

ds2 = ref.get()
ds2.rows[1]['a'] = 42
ref2 = weave.publish(ds2)

ds3 = ref2.get()
assert ds3.rows[1]['a'] == 42
```

Reply:
I don't think the table interface supports that mutation -- only append, pop, insert.  Do we want to add "cell editing" mutations?

Reply:
The existence of server-side table update operation is an implementation detail. We should be thinking about the user patterns that we want to support. 

You might claim that we should explicitly error when setting the "cell", or perhaps we cleanly support it. But users will definitely write this kind of code.

If you want to support with existing apis, a cell set is just a pop followed by an insert.

Anyhow, i think the thing to keep in mind here is that our goal is to correctly track the data for the user (as opposed to directly mapping the underlying mutation logic). In their mind, `rows` is just a list of dicts. The underlying updates/mutations are current implementation details / faculties we have available, but not something the user cares about.